File: backend/src/main/java/com/quickbite/quejas/service/UsuarioService.java
package com.quickbite.quejas.service;

import com.quickbite.quejas.dto.usuario.CambiarEstadoRequest;
import com.quickbite.quejas.dto.usuario.UsuarioPersonalRequest;
import com.quickbite.quejas.dto.usuario.UsuarioResponse;

import java.util.List;

/**
 * CU04 - Administrar Cuentas de Usuario.
 * Alcance: unicamente personal interno (Agente, Supervisor, Administrador).
 * Las cuentas de Cliente se crean por autorregistro (CU02); aqui solo se consultan
 * y se les puede cambiar el estado por moderacion.
 */
public interface UsuarioService {
    List<UsuarioResponse> listarPersonal(String texto, String rol, String estado, Long sucursalId);
    UsuarioResponse crearPersonal(UsuarioPersonalRequest request);
    UsuarioResponse actualizarPersonal(Long id, UsuarioPersonalRequest request);
    UsuarioResponse cambiarEstadoPersonal(Long id);

    List<UsuarioResponse> listarClientes(String texto, String estado);
    UsuarioResponse cambiarEstadoCliente(Long id, CambiarEstadoRequest request);
}

File: frontend/src/app/core/services/usuario.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { UsuarioResponse, UsuarioPersonalRequest, CambiarEstadoRequest } from '../models/usuario.model';

/**
 * CU04 - Administrar Cuentas de Usuario.
 * Alcance: solo personal interno se crea/edita aqui; clientes solo se consultan/suspenden.
 */
@Injectable({ providedIn: 'root' })
export class UsuarioService {
  private readonly baseUrl = `${environment.apiUrl}/usuarios`;

  constructor(private http: HttpClient) {}

  listarPersonal(texto?: string, rol?: string, estado?: string, sucursalId?: number): Observable<UsuarioResponse[]> {
    let params = new HttpParams();
    if (texto) params = params.set('texto', texto);
    if (rol) params = params.set('rol', rol);
    if (estado) params = params.set('estado', estado);
    if (sucursalId) params = params.set('sucursalId', sucursalId);
    return this.http.get<UsuarioResponse[]>(`${this.baseUrl}/personal`, { params });
  }

  crearPersonal(request: UsuarioPersonalRequest): Observable<UsuarioResponse> {
    return this.http.post<UsuarioResponse>(`${this.baseUrl}/personal`, request);
  }

  actualizarPersonal(id: number, request: UsuarioPersonalRequest): Observable<UsuarioResponse> {
    return this.http.put<UsuarioResponse>(`${this.baseUrl}/personal/${id}`, request);
  }

  cambiarEstadoPersonal(id: number): Observable<UsuarioResponse> {
    return this.http.patch<UsuarioResponse>(`${this.baseUrl}/personal/${id}/estado`, {});
  }

  listarClientes(texto?: string, estado?: string): Observable<UsuarioResponse[]> {
    let params = new HttpParams();
    if (texto) params = params.set('texto', texto);
    if (estado) params = params.set('estado', estado);
    return this.http.get<UsuarioResponse[]>(`${this.baseUrl}/clientes`, { params });
  }

  cambiarEstadoCliente(id: number, request: CambiarEstadoRequest): Observable<UsuarioResponse> {
    return this.http.patch<UsuarioResponse>(`${this.baseUrl}/clientes/${id}/estado`, request);
  }
}
